site/src/components/Pill/Pill.tsx

```diff
@@ -44,61 +44,71 @@ const themeStyles =
     };
   };
 
-export const Pill: FC<PillProps> = (props) => {
-  const { lightBorder, icon, text = null, type = "neutral", ...attrs } = props;
-  const theme = useTheme();
+export const Pill: FC<PillProps> = forwardRef<HTMLDivElement, PillProps>(
+  (props, ref) => {
+    const {
+      lightBorder,
+      icon,
+      text = null,
+      type = "neutral",
+      ...attrs
+    } = props;
+    const theme = useTheme();
 
-  const typeStyles = useMemo(() => {
-    if (type in themeOverrides) {
-      return themeOverrides[type as keyof typeof themeOverrides](lightBorder);
-    }
-    return themeStyles(type, lightBorder);
-  }, [type, lightBorder]);
+    const typeStyles = useMemo(() => {
+      if (type in themeOverrides) {
+        return themeOverrides[type as keyof typeof themeOverrides](lightBorder);
+      }
+      return themeStyles(type, lightBorder);
+    }, [type, lightBorder]);
 
-  return (
-    <div
-      css={[
-        {
-          display: "inline-flex",
-          alignItems: "center",
-          borderWidth: 1,
-          borderStyle: "solid",
-          borderRadius: 99999,
-          fontSize: 12,
-          color: "#FFF",
-          height: theme.spacing(3),
-          paddingLeft: icon ? theme.spacing(0.75) : theme.spacing(1.5),
-          paddingRight: theme.spacing(1.5),
-          whiteSpace: "nowrap",
-          fontWeight: 400,
-        },
-        typeStyles,
-      ]}
-      role="status"
-      {...attrs}
-    >
-      {icon && (
-        <div
-          css={css`
-            margin-right: ${theme.spacing(0.5)};
-            width: ${theme.spacing(1.75)};
-            height: ${theme.spacing(1.75)};
-            line-height: 0;
-            display: flex;
-            align-items: center;
-            justify-content: center;
-
-            & > img,
-            & > svg {
+    return (
+      <div
+        ref={ref}
```

Comment:
I am probably missing something obvious, but what does passing in `ref` here do?

Reply:
The ref gives any parent component the ability to manipulate the the div element created by Pill. 

So for example:
```tsx
function Parent () {
  const divRef = useRef<HTMLDivElement>(null);
  return <Pill ref={divRef} {...otherProps} />
}
```
With this setup, the parent would be able to manipulate the div in effects and event handlers, even though it technically exists in a separate component

Reply:
But judging from this PR, I think it's a preemptive change to make the component more reusable. I don't see any of the `<Pill />` instances in the rest of the code using it

Reply:
Yes, it is used by the `Tooltip` component that wraps the `Pill`

Reply:
Ohhhhh I see.  I see it in the docs.  Unfortunate that it is not reflected in the code!  I guess you just have to know how MUI tooltip works.  Thanks for the explanations!